integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -816,10 +816,20 @@ object CarbonDataRDDFactory {
       val partitionByRdd = keyRDD.partitionBy(
         new SegmentPartitioner(segmentIdIndex, segmentUpdateParallelism))
 
+      val conf = SparkSQLUtil.broadCastHadoopConf(sqlContext.sparkSession.sparkContext, hadoopConf)
+      val carbonSessionInfo: CarbonSessionInfo = {
```

Comment:
Instead of this code block from 820-827lines, could have just used 
`carbonSessionInfo = ThreadLocalSessionInfo.getCarbonSessionInfo`

As this object has configuration object  within it, i think, it is better to broadcast and use it.